src/components/legend/defaults.js

```diff
@@ -57,6 +57,7 @@ module.exports = function legendDefaults(layoutIn, layoutOut, fullData) {
     coerce('bordercolor');
     coerce('borderwidth');
     Lib.coerceFont(coerce, 'font', layoutOut.font);
+    coerce('orientation');
```

Comment:
We should add some _smart default_ logic so that if a user set `orientation` to `'h`', the default `'x'` would be `0` (instead of `1.02`).

Moreover, maybe we should make the default `'y'` value `0` instead of `1` ? But that one is less obvious to me. 

@mdtusz @chriddyp @jackparmer should the _default_ horizontal legend start at the bottom-left or at the top-left corner of the plot area?